src/compute/boolean.rs

```diff
@@ -133,3 +134,53 @@ pub fn is_not_null(input: &dyn Array) -> BooleanArray {
     };
     BooleanArray::from_data(DataType::Boolean, values, None)
 }
+
+/// Performs `AND` operation on an array and a scalar value. If either left or right value
+/// is null then the result is also null.
+/// # Example
+/// ```rust
+/// use arrow2::array::BooleanArray;
+/// use arrow2::compute::boolean::and_scalar;
+/// use arrow2::scalar::BooleanScalar;
+/// # fn main() {
+/// let array = BooleanArray::from_slice(&[false, false, true, true]);
+/// let scalar = BooleanScalar::new(Some(true));
+/// let result = and_scalar(&array, &scalar);
+/// assert_eq!(result, BooleanArray::from_slice(&[false, false, true, true]));
+/// # }
+/// ```
+pub fn and_scalar(array: &BooleanArray, scalar: &BooleanScalar) -> BooleanArray {
+    match scalar.value() {
+        Some(true) => array.clone(),
+        Some(false) => {
+            let values = Bitmap::from_trusted_len_iter(std::iter::repeat(false).take(array.len()));
```

Comment:
```suggestion
            let values = Bitmap::from_len_zeroed(array.len());
```

(`from_len_zeroed` uses the equivalent of C's `calloc`, while `from_trusted_len_iter` uses `malloc` followed by setting bit by bit, which is far more expensive ;) )

Reply:
Got it. I will update it with `let values = Bitmap::new_zeroed(array.len());`. Thanks.